torch/linalg/__init__.py

```diff
@@ -482,7 +482,8 @@
         - **solution** (*Tensor*): the least squares solution
         - **residuals** (*Tensor*):  if :math:`m > n` then for full rank matrices in :attr:`input` the tensor encodes
             the squared residuals of the solutions, that is :math:`||\text{input} @ x - b||_F^2`.
-            If :math:`m \le n`, an empty tensor is returned instead.
+            If :math:`m \le n` and rank is not equal to :math:`n`, an empty tensor is returned instead.
```

Comment:
I do not understand this line. I assume that residuals only make sense when the map is not surjective. As such, wouldn't it be much clearer to write in the previous line ``:math:`m \geq n` `` and in this one simply:
```
If :math:`m < n`, an empty tensor is returned instead.
```
Also, the current docs does not say what happens when ``:math:`m \geq n` `` and the matrix is not full-rank.

Reply:
It should be changed to "If m <= n **_or_** rank != n (if m > n), an empty tensor is returned".
I think it's better to avoid computing the residuals, we only do this because it's available from LAPACK for stated conditions and NumPy provides it as well.